internal/provider/resources/install/ssh/aws.go

```diff
@@ -0,0 +1,191 @@
+	data := internal.Configure(&req, resp)
+	r.installer = &installresources.Install{
+		Integration:  SshKey,
+		Component:    installresources.IamWrite,
+		ProviderData: data,
+		GetId:        r.getId,
+		GetItemJson:  r.getItemJson,
+		FromJson:     r.fromJson,
+		ToJson:       r.toJson,
+	}
+}
+
+func (r *AwsSshIamWrite) getId(data any) *string {
+	model, ok := data.(*awsSshIamWriteModel)
+	if !ok {
+		return nil
+	}
+
+	str := fmt.Sprintf("%s%s", awsPrefix, model.AccountId.ValueString())
+	return &str
+}
+
+func (r *AwsSshIamWrite) getItemJson(json any) any {
+	inner, ok := json.(*awsSshIamWriteApi)
+	if !ok {
+		return nil
+	}
+	return inner.Item
+}
+
+func (r *AwsSshIamWrite) fromJson(id string, json any) any {
+	data := awsSshIamWriteModel{}
+	jsonv, ok := json.(*awsSshIamWriteJson)
+	if !ok {
+		return nil
+	}
+
+	// remove the aws prefix.
+	accountId := strings.TrimPrefix(id, awsPrefix)
+	data.AccountId = types.StringValue(accountId)
+	data.Label = types.StringNull()
+	if jsonv.Label != nil {
+		label := types.StringValue(*jsonv.Label)
+		data.Label = label
+	}
+
+	data.State = types.StringValue(jsonv.State)
+	data.GroupKey = types.StringNull()
+	if jsonv.GroupKey != nil {
+		group := types.StringValue(*jsonv.GroupKey)
+		data.GroupKey = group
+	}
+
+	return &data
+}
+
+func (r *AwsSshIamWrite) toJson(data any) any {
+	json := awsSshIamWriteJson{}
+
+	datav, ok := data.(*awsSshIamWriteModel)
+	if !ok {
+		return nil
+	}
+
+	if !datav.Label.IsNull() && !datav.Label.IsUnknown() {
+		label := datav.Label.ValueString()
+		json.Label = &label
+	}
+
+	if !datav.GroupKey.IsNull() {
+		group := datav.GroupKey.ValueString()
+		json.GroupKey = &group
+	}
+
+	// can omit state here as it's filled by the backend
+	return &json
+}
+
+// Create implements resource.ResourceWithImportState.
+func (s *AwsSshIamWrite) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var json awsSshIamWriteApi
+	var data awsSshIamWriteModel
+
+	s.installer.EnsureConfig(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &json, &data)
+	s.installer.Stage(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &json, &data)
+	s.installer.UpsertFromStage(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &json, &data)
+}
+
+func (s *AwsSshIamWrite) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	s.installer.Read(ctx, &resp.Diagnostics, &resp.State, &awsSshIamWriteApi{}, &awsSshIamWriteModel{})
+}
+
+func (s *AwsSshIamWrite) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	s.installer.Delete(ctx, &resp.Diagnostics, &req.State, &awsSshIamWriteModel{})
+}
+
+// Update implements resource.ResourceWithImportState.
+func (s *AwsSshIamWrite) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	s.installer.UpsertFromStage(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &awsSshIamWriteApi{}, &awsSshIamWriteModel{})
```

Comment:
Do you need to `Stage` here?

Reply:
I don't think it's necessary. `Stage` is only being used to make sure the config exists, after that we can perform the other two steps: Verify/Configure safely.